File: csharp/SortBenchmark.cs
using System.Text;

public abstract class SortBenchmark : Benchmark
{
    protected int[] _data;
    protected long _size;
    protected uint _result;

    protected SortBenchmark()
    {
        _data = Array.Empty<int>();
        _result = 0;
        _size = ConfigVal("size");
    }

    public override void Prepare()
    {
        _data = new int[_size];
        for (int i = 0; i < _size; i++) _data[i] = Helper.NextInt(1000000);
    }

    protected abstract int[] Test();

    public override void Run(long IterationId)
    {
        _result += (uint)_data[Helper.NextInt((int)_size)];
        int[] t = Test();
        _result += (uint)t[Helper.NextInt((int)_size)];
    }

    public override uint Checksum => _result;
    public override string TypeName => "Sort";
}

File: swift/Sources/benchmarks/SortBenchmark.swift
import Foundation

class SortBenchmark: BenchmarkProtocol {
  var data: [Int] = []
  private var sizeVal: Int64 = 0
  private var resultVal: UInt32 = 0

  init() {

  }

  func prepare() {
    if sizeVal == 0 {
      sizeVal = configValue("size") ?? 0
      data.reserveCapacity(Int(sizeVal))
      for _ in 0..<Int(sizeVal) {
        data.append(Helper.nextInt(max: 1_000_000))
      }
    }
  }

  func run(iterationId: Int) {
    resultVal &+= UInt32(data[Helper.nextInt(max: Int(sizeVal))])
    let t = test()
    resultVal &+= UInt32(t[Helper.nextInt(max: Int(sizeVal))])
  }

  func test() -> [Int] {
    return []
  }

  var checksum: UInt32 {
    return resultVal
  }

  func name() -> String {
    return ""
  }
}
